modules/feature_home/test/src/domain/repositories/anime_repository_test.dart

```diff
@@ -0,0 +1,408 @@
+import 'package:flutter_test/flutter_test.dart';
+
+import '../../../helper_models.dart';
+
+class AnimeRemoteDataSourceMock extends Mock implements AnimeRemoteDataSource {}
+
+class AnimeCacheDataSourceMock extends Mock implements AnimeCacheDataSource {}
+
+void main() {
+  late AnimeRemoteDataSourceMock animeRemoteDataSourceMock;
+  late AnimeCacheDataSourceMock animeCacheDataSourceMock;
+  late AnimeRepository animeRepository;
+
+  setUp(() {
+    animeRemoteDataSourceMock = AnimeRemoteDataSourceMock();
+    animeCacheDataSourceMock = AnimeCacheDataSourceMock();
+    animeRepository = AnimeRepositoryImpl(
+      remoteDataSource: animeRemoteDataSourceMock,
+      cacheDataSource: animeCacheDataSourceMock,
+    );
+  });
+
+  // This is necessary to make the any() matcher work with the
+  // AnimeDetailsAuxiliarCache type.
+  setUpAll(() {
+    registerFallbackValue(
+      mockAnimeDetailsAuxiliarCache,
+    );
+  });
+
+  group(
+    'getAnimeDetails',
+    () {
+      test(
+        'should return an [AnimeDetails] when the call to remote data source is successful',
+        () async {
+          when(() => animeRemoteDataSourceMock.getAnimeDetails(any()))
+              .thenAnswer((_) async => Result.success(mockAnimeDetails));
+          when(() => animeCacheDataSourceMock.saveAnimeDetails(any()))
+              .thenAnswer((_) async {});
+          when(() => animeCacheDataSourceMock.verifyIfAnimeIsFavorite(any()))
+              .thenAnswer((_) async => false);
+
+          final rawResult = await animeRepository.getAnimeDetails(1);
+
+          final result = rawResult.whenOrNull(
+            success: (animeDetails) => animeDetails,
+          );
+
+          expect(result, equals(mockAnimeDetails));
+
+          verify(() => animeRemoteDataSourceMock.getAnimeDetails(1)) //
+              .called(1);
+          verify(() => animeCacheDataSourceMock.saveAnimeDetails(any()))
+              .called(1);
+          verify(() => animeCacheDataSourceMock.verifyIfAnimeIsFavorite(1))
+              .called(1);
+        },
+      );
+
+      test(
+        'should return an [Exception] when the call to remote data source and cache data source is unsuccessful',
+        () async {
+          final mockException = Exception();
+
+          when(() => animeRemoteDataSourceMock.getAnimeDetails(any()))
+              .thenAnswer((_) async => Result.error(mockException));
+
+          when(() => animeCacheDataSourceMock.getAnimeDetails(any()))
+              .thenAnswer((_) async => Result.error(mockException));
+
+          final rawResult = await animeRepository.getAnimeDetails(1);
+
+          final result = rawResult.whenOrNull(
+            error: (exception) => exception,
+          );
+
+          expect(result, equals(mockException));
+
+          verify(() => animeRemoteDataSourceMock.getAnimeDetails(1)) //
+              .called(1);
+          verify(() => animeCacheDataSourceMock.getAnimeDetails(1)) //
+              .called(1);
```

Comment:
pode deletar o //

Reply:
ah, isso foi só pra forçar a quebra de linha mesmo rsrs.